### PR TITLE
V2.0 jazzcat firefox tests

### PR DESCRIPTION
**Pull request by jansepar**
Gets the Jazzcat tests working in newer Firefox browsers.

### CODE REVIEW

File: tests/jazzcat.html [hunk 1]
```diff
@@ -23,7 +23,17 @@
             require(["../src/jazzcat.js"], function(Jazzcat) {
                 Jazzcat.httpCache.load();
                 var scripts = document.getElementsByTagName("textarea")[0].textContent;
-                document.write(scripts);
+                var temp = document.createElement("div");
+                temp.innerHTML = scripts;
+                document.write("<html><head></head></html>")
+                for (var i=0; i < temp.childNodes.length; i++) {
+                    var scriptEl = temp.childNodes[i];
+                    if (scriptEl.nodeName !== "SCRIPT") return;
```
Comment (noahadams) [inline tests/jazzcat.html]: wat?

Comment (jansepar) [inline tests/jazzcat.html]: There are a bunch of text nodes in here somehow.

Comment (noahadams) [inline tests/jazzcat.html]: This test module explicitly includes jquery, why not use it? This iteration is super ugly, and we're in moderately ugly country here.

Comment (jansepar) [inline tests/jazzcat.html]: I guess I'm just so used to writing without jQuery these days o_O

File: tests/jazzcat.html [hunk 2]
```diff
@@ -37,10 +47,20 @@
             require(["../src/jazzcat.js"], function(Jazzcat) {
                 Jazzcat.httpCache.load();
                 var scripts = document.getElementsByTagName("textarea")[0].textContent;
-                document.write(scripts);
+                var temp = document.createElement("div");
+                temp.innerHTML = scripts;
+                document.write("<html><head></head></html>")
```
Comment (noahadams) [inline tests/jazzcat.html]: really? this seems like this is one step too many of document.write inception and unnecessary. this whole fixture is _already_ being document.written

Comment (noahadams) [inline tests/jazzcat.html]: So, there are two options: simplify this in terms of document.write inception, this is stupidly complicated just to work around async stuff in require... by blowing away the document again.

We could replace this with almond to give us working define/require without require.js actually doing async dynamic file loading.

File: tests/jazzcat.html [hunk 3]
```diff
@@ -15,49 +15,62 @@
...
-            require(["../src/jazzcat.js"], function(Jazzcat) {
-                Jazzcat.httpCache.load();
-                var scripts = document.getElementsByTagName("textarea")[0].textContent;
-                document.write(scripts);
-                parent.postMessage(a, "*");
-            }, undefined, true);
+            Jazzcat.httpCache.load();
+            var scripts = document.getElementsByTagName("textarea")[0].textContent;
+            var temp = document.createElement("div");
+            temp.innerHTML = scripts;
```
Comment (noahadams) [inline tests/jazzcat.html]: bring back document.write(scripts), this eval stuff is bunk!

File: tests/jazzcat.html [hunk 4]
```diff
@@ -15,49 +15,44 @@
 
 <div id="qunit-fixture">
 
-    <textarea id="requirejs-fixture-1">
+    <textarea id="fixture-1">
+        <!-- Bring in Mobify.js to synchronously load jazzcat -->
+        <script src="/build/mobify.js"></script>
```
Comment (noahadams) [inline tests/jazzcat.html:20]: innovative!

**Comment by noahadams** [comment]
:+1: passes on firefox, tests are only crazy to required degree.

**Pull request merged**